src/backend/distributed/planner/function_call_delegation.c

```diff
@@ -230,11 +229,10 @@ TryToDelegateFunctionCall(DistributedPlanningContext *planContext)
 	}
 
 	/*
-	 * This can be called while executing INSERT ... SELECT func(). insert_select_executor
-	 * doesn't get the planned subquery and gets the actual struct Query, so the planning
-	 * for these kinds of queries happens at the execution time.
+	 * Cannot delegate functions for INSERT ... SELECT func(), since they require
+	 * coordinated transactions.
 	 */
-	if (ExecutingInsertSelect())
+	if (PlanningInsertSelect())
```

Comment:
Unfortunately we cannot do `InsertSelectIntoCitusTable(planningContext->originalQuery)` here, since we reach here while doing the `pg_plan_query()` call for the SELECT part of query, which is agnostic about the fact it is part of a bigger INSERT/SELECT.

Reply:
Optional, but I think we could look at planner level alone and remove this condition.